bip-annex.mediawiki

```diff
@@ -0,0 +1,173 @@
+<pre>
+  BIP: XXX
+  Layer: Consensus (soft fork)
+  Title: Taproot Annex Format
+  Author:
+  Comments-Summary: No comments yet.
+  Comments-URI:
+  Status: Draft
+  Type: Standards Track
+  Created: 757967
+  License: BSD-3-Clause
+  Requires: 340, 341, 342
+</pre>
+
+== Introduction ==
+
+=== Abstract ===
+
+This BIP describes a validation format for the taproot annex ([https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki BIP341]).
+It allows to extend the usual transaction fields with new data records allowing witness signatures to commit to them.
+The data records can be subject to new validation rules.
+
+=== Copyright ===
+
+This document is licensed under the 3-clause BSD license.
+
+=== Motivation ===
+
+From the limited set of Bitcoin transaction fields (i.e nVersion, inputs, outputs, nLocktime, etc)
+released in the early days of the network, few soft-forks occurred extending the validation semantic
+of some transaction fields (e.g [https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki BIP68])
+or adding whole new field to solve the malleability issue (e.g [https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki BIP141]). 
+While a generic mechanism consensus to extend the block commmitments have been provisioned with BIP141, 
+there is lacking an equivalent generic mechanism to extend the transaction data fields.
+
+This proposal introduces a format to add new data fields in the Taproot annex. BIP341 mandates
+that if a witness includes at least two elements and the first byte of the last element is 0x50,
+this element is qualified as the annex. The remaining bytes semantics are defined by new validation
+rules following a highly byte efficient Type-Length-Value format.
+
+Specific semantics for the new data fields can be introduced with future soft-forks to enable a range
+of use-cases. For now there is only one nLocktime field in a transaction and all inputs must share
+the same value. It could be possible to define per-input lock-time enabling aggregation of off-chain
+protocols transactions (e.g [https://github.com/lightning/bolts/blob/master/03-transactions.md#htlc-timeout-and-htlc-success-transactions Lightning HTLC-timeout]).
+A commitment to historical block hash could be also a new annex data field to enable replay protection
+in case of persisting forks. Another use-case, a group of input-outputs could be bundled and signed
+together to enable fee-bumping batching of off-chain protocols transactions. <ref> '''What if the
+use-cases require access to the annex fields by Script operations ?''' A new PUSH_ANNEX_RECORD could be
+defined to make accessible annex fields to Script operations. </ref> Beyond, the annex format aims to
+be reusable across spends of SegWit versions.
+
+== Specification ==
+
+=== CompressedInt Integer Encoding ===
```

Comment:
Any historical precedence for this kind of encoding? If so, please add reference.

Reply:
This encoding is from https://github.com/bitcoin/bitcoin/pull/1677/commits/4d6144f97faf9d2a6c89f41d7d2360f21f0b71e2 

I don't think there's any precedence for it in other BIPs

Reply:
This is [unsigned LEB128](https://en.wikipedia.org/wiki/LEB128)?